dartagnan/src/main/java/com/dat3m/dartagnan/program/event/Tag.java

```diff
@@ -73,6 +73,14 @@ private C11() {}
         public static final String MO_ACQUIRE_RELEASE   = "ACQ_REL";
         public static final String MO_SC                = "SC";
 
+        public static String extractStoreMo(String cMo) {
+            return cMo.equals(C11.MO_ACQUIRE_RELEASE) ? C11.MO_RELEASE : cMo;
+        }
+
+        public static String extractLoadMo(String cMo) {
+            return cMo.equals(C11.MO_ACQUIRE_RELEASE) ? C11.MO_ACQUIRE : cMo ;
+        }
+
```

Comment:
I think you need to handle the case of extracting `storeMo` from `acq` and vice versa.
I mean that `extractStoreMo(MO_ACQUIRE)` and `extractLoadMo(MO_RELEASE)` should probably result in `MO_RELAXED`.
Otherwise, the RMW compilation schemes end up producing acquire writes and release loads which don't make sense I think.